pkg/controller/worker/machines.go

```diff
@@ -111,14 +112,17 @@ func (w *workerDelegate) generateMachineConfig() error {
 			return err
 		}
 
-		ami, err := w.findMachineImage(pool.MachineImage.Name, pool.MachineImage.Version, w.worker.Spec.Region)
+		arch := pointer.StringDeref(pool.Architecture, v1beta1constants.ArchitectureAMD64)
```

Comment:
Nice, I like this :) Last comment: Can you add a test case for this, i.e. have a worker pool without the `architecture`? Maybe you can just remove https://github.com/gardener/gardener-extension-provider-aws/pull/565/files#diff-12a1fa4ba4f4250731a53989c476c9407e832ccecae17ad3fc1b2eee100ad7c3R431?

Reply:
Yes removing this line implicitly test the required behavior.